src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/StringBuilderProcessOutputLineListener.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.internal.process;
+
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessOutputLineListener;
+
+public class StringBuilderProcessOutputLineListener implements ProcessOutputLineListener {
```

Comment:
I'm not sure I follow. the client should just need to call sdk.getVersion() and nothing else. Likewise for getComponents() (as I think it's happening now).
If you really want to use this, I think the class should not be public but package-protected (likewise for WaitingProcessOutputLineListener). But I don't feel strongly about this.